### PR TITLE
Add vault

### PR DESCRIPTION
**Pull request by zhenik**

### CODE REVIEW

**Comment by zhenik** [comment]
Some notes:
- `vault server dev` - uses `inmem` (in-memory) backend. [reference](https://github.com/hashicorp/vault/issues/2265), so we need to run vault server, init and useal vault, if we want to use consul as a backend

**Comment by fredrikhgrelland** [comment]
@zhenik why would we want persistent backend? consul and nomad are also inmem.

**Comment by zhenik** [comment]
> @zhenik why would we want persistent backend? consul and nomad are also inmem.

I thought we need to have integration Vault with Consul?

**Comment by fredrikhgrelland** [comment]
> > @zhenik why would we want persistent backend? consul and nomad are also inmem.
> 
> I thought we need to have integration Vault with Consul?
 Yes, but as a secret backend. ( meaning vault will handle ACL ) https://www.vaultproject.io/docs/secrets/consul

**Comment by zhenik** [comment]
So, vault `dev` mode will be sufficient then for our needs?

File: ansible/templates/vault.service.j2 [hunk 1]
```diff
@@ -0,0 +1,18 @@
+[Unit]
+Description=vault agent
+Requisite=consul.service
+After=consul.service
+
+[Service]
+EnvironmentFile=-/etc/sysconfig/vault
+Environment=GOMAXPROCS=2
+Restart=on-failure
+ExecStartPre=/bin/sleep 35
```
Comment (fredrikhgrelland) [inline ansible/templates/vault.service.j2:10]: Is this necessary?

File: ansible/templates/vault.hcl.j2 [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+ui = true
+disable_mlock = true
+
+api_addr      = "http://127.0.0.1:8200"
+cluster_addr  = "http://127.0.0.1:8201"
+
+storage "consul" {
+    address = "127.0.0.1:8500"
+    token = "b6e29626-e23d-98b4-e19f-c71a96fbdef7"
+}
```
Comment (fredrikhgrelland) [inline ansible/templates/vault.hcl.j2:12]: Let us skip the storage backend.

Comment (fredrikhgrelland) [inline ansible/templates/vault.hcl.j2:12]:
```suggestion
```

File: ansible/templates/vault.hcl.j2 [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+listener "tcp" {
+    address     = "127.0.0.1:8200"
+    tls_disable = 1
+}
```
Comment (fredrikhgrelland) [inline ansible/templates/vault.hcl.j2:17]:
```suggestion
listener "tcp" {
  address = "{{ GetInterfaceIP \"docker0\" }}:8200"
}
listener "tcp" {
  address = "{{ GetInterfaceIP \"eth0\" }}:8200"
}
listener "tcp" {
  address = "{{ GetInterfaceIP \"eth1\" }}:8200"
}
```

File: ansible/templates/vault.hcl.j2 [hunk 4]
```diff
@@ -0,0 +1,18 @@
+{% raw -%}
+
+ui = true
+disable_mlock = true
+
+api_addr      = "http://127.0.0.1:8200"
+cluster_addr  = "http://127.0.0.1:8201"
```
Comment (fredrikhgrelland) [inline ansible/templates/vault.hcl.j2:7]:
```suggestion
# Advertise the non-loopback interface ( eth1)
api_addr      = "http://{{ GetInterfaceIP \"eth1\" }}:8200"
cluster_addr  = "http://{{ GetInterfaceIP \"eth1\" }}:8201"
```

File: ansible/templates/vault.hcl.j2 (hunk 3 above)
Comment (zhenik) [reply ansible/templates/vault.hcl.j2:17]: I am not sure that vault support such a configuration parsing
```
Error initializing listener of type tcp: listen tcp: lookup {{ GetInterfaceIP "docker0" }}: no such host
```

**Comment by zhenik** [comment]
ref to update config:
- https://github.com/hashicorp/vault/pull/9109
- https://github.com/hashicorp/vault/issues/6409

**Pull request merged**